packages/task/src/browser/task-service.ts

```diff
@@ -1105,7 +1105,11 @@ export class TaskService implements TaskConfigurationClient {
      * @param task The task to configure
      */
     async configure(token: number, task: TaskConfiguration): Promise<void> {
-        await this.taskConfigurations.configure(token, task);
+        const taskConfiguration = Object.freeze({
```

Comment:
This looks like it's trying to fix the bug where labels were not preserved through this process, but it could have unintended consequences. Were you able to track down the code where the label was being modified or deleted? It might be better to fix the problem there rather than trying to freeze the object at a moment we think we're happy with it.

Reply:
The added lines are not displayed correctly by github, please see the image below for the set of lines added.

if we ignore freeze for a moment, the intention is simply to populate a label under the task configuration, before this fix, the label does not get preserved in the tasks.json file.  The freeze could be replaced by Object.assign.
Although freeze is chosen to make it explicit that the configuration shall then be complete.

![image](https://user-images.githubusercontent.com/76971376/106472309-89720d80-6470-11eb-8676-3bda693d51c6.png)

Reply:
If freezing isn't critical to the intended functionality, I'd suggest removing it and just adding the label here, since `freeze` could produce hard-to-diagnose runtime errors for anyone who works on this code later.

Reply:
Since the access associated to TaskConfiguration.label is 'readonly' we do have the options to add it with Object.create, Object.assign, Object.freeze, if we don't use Object.freeze, I am fine with using assign or create, please let me know if you have a preference.

Reply:
It seems as though the problem is just that the label was being omitted previously - is that correct? If so I believe this should be fine:
```
const taskConfiguration = { ...task, label: this.taskNameResolver.resolve(task) };
```
If there's a deeper problem with labels being manipulated inappropriately downstream of this code, or if they're being omitted inappropriately upstream of this code, then it might be best to find those locations and address the bug there. If the label just wasn't included at this stage of the task construction, then we just need to add it here.

Reply:
The label was not actually omitted at this point, but it was not prefixed with the source, so we call taskNameResolver to apply the prefix, then the actual storing of the label in the tasks.json was missing and it's fixed under task-configuration.ts.

I think your suggestion may work although I think it's best to use Object.assign or Object.create in order to preserve the prototype.

Reply:
I believe that `TaskConfiguration` is just an interface, not a class, so the prototype of the object we're dealing with is just the global object type, so destructuring and assigning the label won't change the prototype, but if you prefer one of the `Object` methods, by all means `Object.assign(task, {label: this.taskNameResolver.resolve(task)}`.

Reply:
Done, on a new version of the commit. Thanks !